### PR TITLE
Add GetSemanticModel method with disableNullableAnalysis flag

### PR DESCRIPTION
**Pull request by RikkiGibson**
Related to #70609

We'd like to do a "spike" of a simple implementation of this API, to find out if we are getting perf improvements in classification in practice. If that works out, we'll want to revisit what the actual API shape should be and also further scrutinize the implementation.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs [hunk 1]
```diff
@@ -104,6 +104,15 @@ public sealed override bool IgnoresAccessibility
             }
         }
 
+        // PROTOTYPE(ndsm): it feels strange to have both 'DisableNullableAnalysis' and 'IsNullableAnalysisEnabled' members. But maybe it's fine.
```
Comment (jjonescz) [inline src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs:107]: Maybe `IsNullableAnalysisEnabledCore` could be named something like `CanNullableAnalysisBeEnabled` so it's clear that it's not the final thing, but must be combined with the `DisableNullableAnalysis` flag.

**Comment by maryamariyan** [comment]
Are the CI failures on the PR intermittent? I don't have permission to rerun.

**Comment by CyrusNajmabadi** [comment]
I also see lots of legitimate test failures.

**Comment by RikkiGibson** [comment]
It appears this change is overflowing the stack in some tests. Will try and look into it this afternoon.

**Comment by RikkiGibson** [comment]
Changing base to `demos/...` to reflect that we don't expect the usual level of scrutiny for feature development here. Once CI finishes I will merge.

After that point @CyrusNajmabadi do you have what you need to send a subsequent PR to start using the API in the classifier?

**Comment by CyrusNajmabadi** [comment]
@RikkiGibson Yes.  But @maryamariyan Is going to be doing the work (esp. as she has the work to profile things and validate the performance benefits between the old and new approach).

**Comment by RikkiGibson** [comment]
go for it!

**Pull request merged**